internal/client/blueprints_test.go

```diff
@@ -1229,3 +1229,66 @@ func TestBlueprintChangeV1(t *testing.T) {
 	require.NotNil(t, bp, "GET blueprint change failed: missing blueprint")
 	assert.Equal(t, bp.Version, "0.0.1")
 }
+
+func TestEmptyPackageNameBlueprintJsonV0(t *testing.T) {
+	for _, tc := range []struct {
+		packageSnippet string
+		specificErr    string
+	}{
+		{
+			`"packages": [{"name": "", "version": "*"}]`,
+			"Entry #1 has version '*' but no name.",
+		}, {
+			`"packages": [{"name": ""}]`,
+			"Entry #1 has no name.",
+		},
+	} {
+		bp := `{
+ 		"name": "test-emptypackage-blueprint-v0",
+ 		"description": "TestEmptyPackageNameBlueprintV0",
+ 		"version": "0.0.1",
+		` + tc.packageSnippet + `}`
+
+		expectedErrorPrefix := "BlueprintsError: All package entries need to contain the name of the package."
+		expectedErr := expectedErrorPrefix + " " + tc.specificErr
+
+		resp, err := PostJSONBlueprintV0(testState.socket, bp)
+		require.NoError(t, err, "failed with a client error")
+		require.False(t, resp.Status, "Negative status expected.")
+		require.Equal(t, len(resp.Errors), 1, "There should be exactly one error")
+		reasoning := "I expected the error message\n" + expectedErr +
+			"\nnot\n" + resp.Errors[0].String()
+		assert.True(t, strings.HasPrefix(resp.Errors[0].String(), expectedErr), reasoning)
```

Comment:
(nitpick) - any specific reason to not just use `assert.Equal()` here? it's very nice and  gives a full diff when things are different (and makes the "reasoning" string obsolete)

Reply:
no, sorry, copy'n'paste and too little experience with go :-/